File: sem2/KP/L1/Zad2/LiczbyPierwsze.cpp
#include<math.h>
#include"LiczbyPierwsze.hpp"
#include<iostream>
int LiczbyPierwsze::liczba(int m){ return this->tab[m]; }
LiczbyPierwsze::LiczbyPierwsze(int n){
    //printf("gen %d\n", n);
    
    int index = 0, i = 2;
    bool prime;
    while(i < n){
        prime = true;
        for(int j = 2; j < i; j++){
            if(i%j == 0){
                prime = false;
                break;
            }
        }
        if(prime){
            this->tab[index] = i;
            index++;
        }
        i++;
    }
    this->max = index;
}
LiczbyPierwsze::~LiczbyPierwsze(){
    printf("dest\n");
    return;
}

File: sem2/KP/L1/Zad1/LiczbyPierwsze.java
package Zad1;

public class LiczbyPierwsze{
    private int[] tab;
    public int liczba(int m){ return tab[m]; }
    LiczbyPierwsze(int n){
        tab = new int[n];
        int index = 0, i = 2;
        boolean p;
        while(i < n){
            p = true;
            for(int j = 2; j <= Math.sqrt(i); j++){
                if(i%j == 0){
                    p = false;
                    break;
                }
            }
            if(p){
                tab[index] = i;
                index++;
            }
            i++;
        }
    }
}
